### PR TITLE
[Project Search] Fix duplicate sources

### PR DESCRIPTION
**Pull request by AlexisDeschamps**
#5159

### Summary of Changes
* Removed `loadAllFiles` method
* Started loading files in `searchSources`

### Test Plan
- Open https://arewesmoothyet.com/?mode=track&trackedStat=Devtools%20Hangs
- CMD-SHIFT-F for "trackedStat"
- Expect one result instead of two

### Screenshots
Before:
image

After:
image

### CODE REVIEW

**Comment by jasonLaster** [comment]
Looks great. I'll review more fully tuesday

File: src/actions/project-text-search.js [hunk 1]
```diff
@@ -48,12 +48,12 @@ export function searchSources(query: string) {
...
     await dispatch(addSearchQuery(query));
     dispatch(updateSearchStatus(statusType.fetching));
-    await dispatch(loadAllSources());
     const sources = getSources(getState());
     const validSources = sources
       .valueSeq()
-      .filter(source => isLoaded(source) && !isThirdParty(source));
+      .filter(source => !isThirdParty(source));
     for (const source of validSources) {
+      await dispatch(loadSourceText(source));
```
Comment (lukaszsobek) [inline src/actions/project-text-search.js]: `loadSourcetext` calls `isLoaded` anyway, nice.

File: src/actions/project-text-search.js [hunk 2]
```diff
@@ -48,12 +48,12 @@ export function searchSources(query: string) {
     await dispatch(clearSearchResults());
     await dispatch(addSearchQuery(query));
     dispatch(updateSearchStatus(statusType.fetching));
-    await dispatch(loadAllSources());
```
Comment (lukaszsobek) [inline src/actions/project-text-search.js]: Like the shift in approach. Instead of loading everything up front we load them one by one.

**Comment by lukaszsobek** [comment]
Maybe we should move `updateSearchStatus(statusType.done)`
from
https://github.com/devtools-html/debugger.html/pull/5165/files#diff-7741e0df431e1dec7a1070517efd281bL79
to 
https://github.com/devtools-html/debugger.html/pull/5165/files#diff-7741e0df431e1dec7a1070517efd281bL59

**Comment by lukaszsobek** [comment]
If we don't already have one it would be great to have a tests that checks if the status is correctly updated after all sources were loaded.

**Comment by jbhoosreddy** [comment]
@lukaszsobek I did the `updateSearchStatus(statusType.done)` refactor in a separate PR #5124. We could leave it there to prevent merge conflicts.

**Comment by jasonLaster** [comment]
Hey @jbhoosreddy, @AlexisDeschamps could you do a rebase here

**Pull request merged**